consensus/spos/bls/subroundSignature.go

```diff
@@ -378,6 +389,12 @@ func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
 			return false
 		}
 
+		isKeyLeader := idx == spos.IndexOfLeaderInConsensusGroup
+		// TODO[cleanup cns finality]: update the check
+		// with the equivalent messages feature on, signatures from all managed keys must be broadcast, as the aggregation is done by any participant
+		if isFlagActive {
+			isMultiKeyLeader = isKeyLeader
+		}
 		if !isMultiKeyLeader {
```

Comment:
i think here it would help to have a separate variable like `shouldBroadcastSignature`, since if `isFlagActive` = true `isMultiKeyLeader` variable name is confusing

Reply:
if we can refactor a bit here, it's a bit difficult to track `isFlagActive` `isMultiKeyLeader` `isKeyLeader` conditions in this func, at least at first

Reply:
renamed a bit for clarity